### PR TITLE
Implemented diagnostic for SA1118: ParameterMustNotSpanMultipleLines 

### LINKED ISSUE #52: SA1118: ParameterMustNotSpanMultipleLines
http://www.stylecop.com/docs/SA1118.html

### PR DESCRIPTION
**Pull request by macpak**
Fixes #52.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1118ParameterMustNotSpanMultipleLines.cs [hunk 1]
```diff
@@ -77,7 +87,38 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
...
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(this.HandleArgumentList, SyntaxKind.ArgumentList, SyntaxKind.BracketedArgumentList);
+        }
+
+        private void HandleArgumentList(SyntaxNodeAnalysisContext context)
+        {
+            var argumentListSyntax = (BaseArgumentListSyntax) context.Node;
+            if (argumentListSyntax.Arguments.Count < 2)
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1118ParameterMustNotSpanMultipleLines.cs]: This is already excluded by the loop

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1118ParameterMustNotSpanMultipleLines.cs [hunk 2]
```diff
@@ -54,7 +57,7 @@ public class SA1118ParameterMustNotSpanMultipleLines : DiagnosticAnalyzer
         /// </summary>
         public const string DiagnosticId = "SA1118";
         private const string Title = "Parameter must not span multiple lines";
-        private const string MessageFormat = "TODO: Message format";
+        private const string MessageFormat = "The parameter spans multiple lines. If the parameter is short, place the entire parameter on a single line. Otherwise, save the contents of the parameter in a temporary variable and pass the temporary variable as a parameter.";
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1118ParameterMustNotSpanMultipleLines.cs]: Do we really want a message this long? The Light bulb is cutting of parts of it
Light bulb

**Comment by pdelvo** [comment]
This rule should be enabled by default.

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1118ParameterMustNotSpanMultipleLines.cs (hunk 2 above)
Comment (macpak) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1118ParameterMustNotSpanMultipleLines.cs]: I used the original StyleCop message, but of course we can have a shorter one.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1118UnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    }
+}";
+
+            DiagnosticResult expected = this.CSharpDiagnostic().WithLocation(11, 13);
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMethodCallWithThreeParametersSecondSpansMultipleLinesThirdSpansMultipleLinesButIsLambdaExpression()
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1118UnitTests.cs]: You could merge this test with the previous one by using a theory

**Comment by pdelvo** [comment]
You might want to add a test for attributes

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1118UnitTests.cs [hunk 4]
```diff
@@ -0,0 +1,252 @@
+﻿namespace StyleCop.Analyzers.Test.ReadabilityRules
+{
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using System.Threading;
+    using StyleCop.Analyzers.ReadabilityRules;
+    using TestHelper;
+    using Xunit;
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1118UnitTests.cs]: The order of these using directives is wrong

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1118ParameterMustNotSpanMultipleLines.cs (hunk 2 above)
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1118ParameterMustNotSpanMultipleLines.cs]: @macpak That's what the `Description` value is for. The message should be short; many times we reuse the title but occasionally we have a more specific message (e.g. 8b503ee).

Comment (macpak) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1118ParameterMustNotSpanMultipleLines.cs]: Yeah, you're right. I will change it. :+1:

**Pull request merged**